compass/Coordinator.java

```diff
@@ -405,16 +425,23 @@ private GetNodeInfoResponse getNodeInfoWithRetries() throws InterruptedException
     return response;
   }
 
-  private GetTransactionsToApproveResponse getGetTransactionsToApproveResponseWithRetries() throws InterruptedException {
+  private GetTransactionsToApproveResponse getGetTransactionsToApproveResponseWithRetries() throws
+          TimeoutException, InterruptedException {
     GetTransactionsToApproveResponse response = null;
     for(int i = 0; i < config.APIRetries; i++) {
       try {
         response = api.getTransactionsToApprove(depth, state.latestMilestoneHash);
         break;
-      } catch (IllegalStateException | ArgumentException | IllegalAccessError e) {
+      } catch (IllegalStateException | IllegalAccessError e) {
         log.error("API call failed: ", e);
         Thread.sleep(config.APIRetryInterval);
       }
+      catch  (ArgumentException e) {
+        log.error("There was a problem processing Get Transactions To Approve: ", e);
+        if (e.getMessage().contains("exceeded timeout")) {
+          throw new TimeoutException("Get Transactions To Approve call timed out", e);
+        }
+      }
```

Comment:
else you should also sleep before a new attempt.
did you test this in practice and get a different Exception type?

Reply:
I did test 
If it is a timeout I don't want to sleep

Reply:
I agree, I'm talking about if it is not a timeout.
I.e. If the error does not contain 'exceeded timeout', the code path now will retry straight away and not sleep like the other no - timeout exceptions

Reply:
fixed